src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -27,17 +27,27 @@ using namespace std::chrono_literals;
 static constexpr short KeyPressed{ gsl::narrow_cast<short>(0x8000) };
 static constexpr auto FrameUpdateInterval = std::chrono::milliseconds(16);
 
-static winrt::com_ptr<IVirtualDesktopManager> s_desktopManager;
+winrt::com_ptr<IVirtualDesktopManager> getDesktopManager()
+{
+    static til::shared_mutex<winrt::com_ptr<IVirtualDesktopManager>> s_desktopManager;
+
+    if (const auto manager = *s_desktopManager.lock_shared())
+    {
+        return manager;

```

Comment:
but this will still return a dead RPC object if we had one stored, right? there's no way to communicate failure and invalidate it

Reply:
I tried it by killing explorer.exe but it still worked for me. Does the client perhaps reconnect automatically?